app/src/main/java/com/sopt/dive/presentation/mypage/MyPageScreen.kt

```diff
@@ -35,34 +43,51 @@ fun MyPageRoute(
 ) {
     val context = LocalContext.current
     val userManager = remember { UserManager(context) }
-    val userData = remember { userManager.loadUserData() }
+    val viewModel : MyPageViewModel = viewModel(
+        factory = ViewModelFactory(
+            authRepository = RepositoryModule.authRepository,
+            userRepository  = RepositoryModule.userRepository,
+            context = context
+        )
+    )
```

Comment:
요렇게 작성을 하게 되면 매번 새로운 Factory 인스턴스 생성해서 Recomposition이 발생할 때마다 ViewModelFactory와 UserManager가 새로 생성될 수 있어욥. remember로 감싸지 않으면 매 recomposition마다 factory가 재생성되어서 ViewModel의 생명주기 이점 활용이 미흡 해질 수 있어요. ViewModel은 Configuration Change에서 살아남아야 하는데, factory를 재생성하는 패턴은 이를 방해할 수 있어요 !! 지금 상황에서는 Factory를 remember로 감싸주는게 좋을 것 같아요
```suggestion
    val viewModel: MyPageViewModel = viewModel(
        factory = remember {
            ViewModelFactory(
                authRepository = RepositoryModule.authRepository,
                userRepository = RepositoryModule.userRepository,
                context = context
            )
        }
    )
```